stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/BackgroundDecorator.kt

```diff
@@ -6,31 +6,62 @@ import android.graphics.Paint
 import android.graphics.Path
 import android.graphics.PixelFormat
 import android.graphics.drawable.Drawable
-import androidx.annotation.ColorInt
 import androidx.core.content.ContextCompat
 import com.getstream.sdk.chat.adapter.MessageListItem
 import com.getstream.sdk.chat.utils.Utils.dpToPx
 import io.getstream.chat.android.ui.R
 import io.getstream.chat.android.ui.messages.adapter.viewholder.MessageDeletedViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.MessagePlainTextViewHolder
 
 internal class BackgroundDecorator : BaseDecorator() {
 
     override fun decorateDeletedMessage(viewHolder: MessageDeletedViewHolder, data: MessageListItem.MessageItem) {
-        val context = viewHolder.itemView.context
-        val backgroundColor = ContextCompat.getColor(context, MESSAGE_DELETED_BACKGROUND)
-        val radius = dpToPx(DEFAULT_CORNER_RADIUS.toInt()).toFloat()
+        val backgroundColor = ContextCompat.getColor(viewHolder.itemView.context, MESSAGE_DELETED_BACKGROUND)
+        val radius = dpToPx(DEFAULT_CORNER_RADIUS_DP).toFloat()
+        val paint = Paint().apply {
+            color = backgroundColor
+            isAntiAlias = true
+            style = Paint.Style.FILL
+        }
         viewHolder.binding.deleteLabel.background = BackgroundDrawable(
-            color = backgroundColor,
+            paint = paint,
             topLeftCornerPx = radius,
             topRightCornerPx = radius,
             bottomRightCornerPx = if (data.positions.contains(MessageListItem.Position.BOTTOM)) 0f else radius,
             bottomLeftCornerPx = radius
         )
     }
 
+    override fun decoratePlainTextMessage(
+        viewHolder: MessagePlainTextViewHolder,
+        data: MessageListItem.MessageItem
+    ) {
+        val radius = dpToPx(DEFAULT_CORNER_RADIUS_DP).toFloat()
+        val paint = if (data.isMine) {
+            Paint().apply {
+                style = Paint.Style.FILL
+                color = ContextCompat.getColor(viewHolder.itemView.context, MESSAGE_CURRENT_USER_BACKGROUND)
+                isAntiAlias = true
+            }
```

Comment:
I wonder if these `Paint` instances would be worth creating in a broader scope, and then re-using. As I understand, we're creating these on each `bind()` in the adapter.

Reply:
I think it's an optimization question. I was always taught not to do pre-optimizations if I'm not sure about the reason.

Reply:
Object creations while binding are the one place where I tend to optimize prematurely, because they happen so often, but alright, it's not a huge priority 👍